lib/werewolf/engine/state.rb

```diff
@@ -0,0 +1,46 @@
+module Werewolf
+  module Engine
+    class State
+      attr_reader :current, :turn_number, :day_number, :enumerator
+
+      STATES = [:dawn, :day, :dusk, :night]
+      TRANSITIONS = [:dawn, :dusk]
+
+      # We start at turn 0, :dawn on day 0
+      def initialize
+        @enumerator = STATES.cycle
+        @current = @enumerator.next
+        @index, @turn_number, @day_number = 0, 0, 0
+      end
+
+      # Advance to the next time period
+      def next
+        # increment the index and turn every time next is called
+        @index += 1
+        @turn_number += 1
+        # only increment the day if a full cycle has happened
+        @day_number += @index/STATES.length
+        # reset index once we have hit then end of a cycle
+        @index = 0 if @index >= STATES.length
+        @current = @enumerator.next
+      end
+
+      def transitioning?
```

Comment:
embrace your inner ruby and convert this from tech-speak to the infinitely more poetic `twilight?`